frontend-public/src/components/modalContent/variances/EditVarianceModal.js

```diff
@@ -0,0 +1,94 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import CustomPropTypes from "@/customPropTypes";
+import EditVarianceForm from "@/components/Forms/variances/EditVarianceForm";
+import {
+  fetchVarianceById,
+  removeDocumentFromVariance,
+  fetchVariancesByMine,
+} from "@/actionCreators/varianceActionCreator";
+import { getVariance } from "@/selectors/varianceSelectors";
+import LoadingWrapper from "@/components/common/wrappers/LoadingWrapper";
+
+const propTypes = {
+  onSubmit: PropTypes.func.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  mineGuid: PropTypes.string.isRequired,
+  mineName: PropTypes.string.isRequired,
+  variance: CustomPropTypes.variance,
+  fetchVarianceById: PropTypes.func.isRequired,
+  varianceGuid: PropTypes.string.isRequired,
+  removeDocumentFromVariance: PropTypes.func.isRequired,
+  varianceStatusOptionsHash: PropTypes.objectOf(PropTypes.string).isRequired,
+  complianceCodesHash: PropTypes.objectOf(PropTypes.string).isRequired,
+  fetchVariancesByMine: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  variance: {},
+};
+
+export class EditVarianceModal extends Component {
+  state = { isLoaded: false };
+
+  componentDidMount() {
+    this.props.fetchVarianceById(this.props.mineGuid, this.props.varianceGuid).then(() => {
+      this.setState({ isLoaded: true });
+    });
+  }
+
+  // handling delete functionality inside the modal, so the data can be updated properly.
```

Comment:
This comment probably makes sense if you know what the alternative approach(es) could be, but coming across the code for the first time, it doesn't tell me anything.

Reply:
removed!